habitTrackerApp.py

```diff
@@ -155,26 +154,31 @@ def set_date(self, widget, *args, **kwargs):
         
         self.populate_header()
 
-        print('we are in set_date method')
         print(widget.background_color)
         print(MyBoxLayout.chosenColor)
+
         if widget.background_color == MyBoxLayout.chosenColor:
             print('we are in white color')
             widget.background_color = [1,1,1,1]
         else:
             print('color is: ' + str(MyBoxLayout.chosenColor))
             widget.background_color = MyBoxLayout.chosenColor
 
-        #Bookmark 25.01.2022: find out why the label text isn't updating automatically.
         counter = 0
         for dates in reversed(MyBoxLayout.buttonList[:MyBoxLayout.buttonList.index(MyBoxLayout.TheTodayButton)+1]):
```

Comment:
```
 reversed(MyBoxLayout.buttonList[:MyBoxLayout.buttonList.index(MyBoxLayout.TheTodayButton)+1]):
```
avoid this long chain in a for loop, it is quite hard to read and understand what is happening.

To make it more readable you can just add it to variable that explains it, without making any changes
```
previous_dates_from_today = reversed(MyBoxLayout.buttonList[:MyBoxLayout.buttonList.index(MyBoxLayout.TheTodayButton)+1]):

for dates in previous_dates_from_today:
```